Week4/woojo230/mission03/src/pages/signup.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState } from 'react';
+import { AnimatePresence, motion } from 'framer-motion';
+import EmailStep from '../components/signup/EmailStep';
+import PasswordStep from '../components/signup/PasswordStep';
+import { emailSchema, signUpSchema } from '../schemas/signupSchema';
+import ProfileStep from '../components/signup/ProfileStep';
+import type { ProfileSteps } from '../type/type';
+import { useNavigate } from 'react-router-dom';
+import { signUpUser } from '../apis/auth/signup';
+
+const SignUp = () => {
+  const [step, setStep] = useState<ProfileSteps>('email');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const navigate = useNavigate();
+
+  // 이메일->패스워드
+  const handleEtoP = (email: string) => {
+    setEmail(email);
+    setStep('password');
+  };
+
+  // 패스워드->프로필
+  const handlePtoP = (password: string) => {
+    setPassword(password);
+    setStep('profile');
+  };
+
+  const handleSignUpComplete = async (
+    nickname: string,
+    profileImage: File | null
+  ) => {
+    try {
+      const data = await signUpUser({
+        email,
+        password,
+        nickname,
+        profileImage,
+      });
+
+      console.log('회원가입 성공:', data);
+      navigate('/home');
```

Comment:
회원가입 성공 시 현재 홈 페이지로 이동되도록 구현되어 있는데,일반적인 회원가입 흐름에서는 가입 성공 후 로그인 페이지로 리다이렉트하는 것이 더 자연스러울 것 같습니다!

```suggestion
      console.log('회원가입 성공:', data);
      navigate('/login');
```